### PR TITLE
Import type for TS

### LINKED ISSUE #10981: `import type` in TypeScript
TypeScript 3.8 will support a new `import type` construct. While there's already some logic for similar stuff (Flow has its own `import type`), TypeScript's has a few extra restrictions.

* `import type` can only be used on imports with named imports or with a default import - but not both.

  ```ts
  import type FooDefault, { Bar, Baz } from "module"; // error!
  ```
* Even though `import type` creates no emit, an `import type` functionally shadows variables from the outer scope, and using these declarations in a value position that will be emitted causes an error.

  ```ts
  import type { RegExp } from "special-regexp-module";

  // Okay - RegExp won't be used at runtime here.
  declare class Foo extends RegExp { }

  // Okay - RegExp won't be used at runtime here.
  let r: typeof RegExp;

  // Error! RegExp is imported via import type and cannot be used as a value here.
  new RegExp()
  ```

See more at

* https://github.com/microsoft/TypeScript/pull/35200
* https://github.com/microsoft/TypeScript/pull/36092
* https://github.com/microsoft/TypeScript/issues/36138

### PR DESCRIPTION
**Pull request by hzoo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | fixes #10981
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      | Yea
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

Implemented for import type, still need to implement export

### CODE REVIEW

File: packages/babel-parser/src/plugins/typescript/index.js [hunk 1]
```diff
@@ -1945,7 +1967,7 @@ export default (superClass: Class<Parser>): Class<Parser> =>
             member.start,
             `Index signatures cannot have an accessibility modifier ('${
               (member: any).accessibility
-            }')`,
+            } ')`,
```
Comment (existentialism) [inline packages/babel-parser/src/plugins/typescript/index.js]:
```suggestion
            }')`,
```

Comment (nicolo-ribaudo) [inline packages/babel-parser/src/plugins/typescript/index.js]:
```suggestion
            }')`,
```

**Comment by nicolo-ribaudo** [comment]
I have changed the base branch to `feat-7.9.0/ts-import-type` so that we can merge it there and work on `export type` and the transform plugin option.

**Comment by nicolo-ribaudo** [comment]
:warning: This PR is not merged to master yet

**Pull request merged**